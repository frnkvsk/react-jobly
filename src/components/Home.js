import React from 'react';
import { makeStyles } from '@material-ui/core/styles';
import Button from '@material-ui/core/Button';


const useStyles = makeStyles((theme) => ({
  root: {
    display: 'flex',
    flexDirection: 'column',
    alignItems: 'center',
    justifyContent: 'center',
    fontSize: '28px',
    height: '100vh',
  }
}));

const Home = () => {
  const classes = useStyles();
  return (
    <div className={classes.root}>
      <h2>Jobly</h2>
      <p>All the jobs in on, convenient place.</p>
      <Button variant="contained" color="primary">
        Log In
      </Button>
    </div>
  );  
}

export default Home;