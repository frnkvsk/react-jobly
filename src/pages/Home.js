import React, { useContext } from 'react';
import { makeStyles } from '@material-ui/core/styles';
import Button from '@material-ui/core/Button';
import { AuthContext } from '../context/AuthContext';

const useStyles = makeStyles((theme) => ({
  root: {
    display: 'flex',
    flexDirection: 'column',
    alignItems: 'center',
    justifyContent: 'flex-start',
    minHeight: '100vh',
    fontSize: '28px',
    height: '100vh',
  },
  main: {
    display: 'flex',
    flexDirection: 'column',
    alignItems: 'center',
    paddingTop: '20vh',
  }
}));

const Home = () => {
  const classes = useStyles();
  const auth = useContext(AuthContext);
  return (
    <div className={classes.root}>
      <main className={classes.main}>
        <h2>Jobly</h2>
        <p>All the jobs in on, convenient place.</p>
        {!auth.authState.token ? 
        <Button variant="contained" color="primary">
          log in / sign up  
        </Button>
        : null}
      </main>
      
    </div>
  );  
}

export default Home;