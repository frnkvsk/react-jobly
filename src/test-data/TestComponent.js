// import React, { useContext } from 'react';
// import { AuthContext } from '../context/AuthContext';

// const TestComponent = () => {
//   const auth = useContext(AuthContext);
//   const setAuth2 = async () =>
//     await auth.setAuthState({
//       token: 'testtoken',
//       userInfo: {
//       username: 'testuser',
//       first_name: 'testfirst',
//       last_name: 'testlast',
//       email: 'test@test.com',
//       photo_url: null
//     }});
//   (async () => await setAuth2())();
//   console.log('-------auth.authState',auth.authState)
//   return (
//     <div>
//       <div>{auth.authState.userInfo.username}</div>
//       <div>{auth.authState.userInfo.first_name}</div>
//       <div>{auth.authState.userInfo.last_name}</div>
//       <div>{auth.authState.userInfo.email}</div>
//       <div>{auth.authState.userInfo.photo_url}</div>
//     </div>
//   )
// }
// export default TestComponent;

