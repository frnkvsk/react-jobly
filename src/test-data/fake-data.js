const testData = {
  jobs: [
    {
    company_handle: "test-company-a",
    equity: 0.115,
    id: 1,
    salary: 11111,
    state: null,
    title: "Editor, magazine features",
    },
      {
      company_handle: "test-company-b",
      equity: 0.22,
      id: 2,
      salary: 22222,
      state: null,
      title: "Coder",
    },
      {
      company_handle: "test-company-c",
      equity: 0.33,
      id: 3,
      salary: 33333,
      state: null,
      title: "Front-end Dev",
    },
  ],
  companies: [
    {
      description: "Test description 1",
      handle: "test-company-1",
      logo_url: "http://www.gtdesigns.it/wp-content/uploads/OverusedLogos/99gen_circle.jpg",
      name: "Test Company 1",
    },
    {
      description: "Test description 2",
      handle: "test-company-2",
      logo_url: "http://www.gtdesigns.it/wp-content/uploads/OverusedLogos/99gen_circle.jpg",
      name: "Test Company 2",
    },
    {
      description: "Test description 3",
      handle: "test-company-3",
      logo_url: "http://www.gtdesigns.it/wp-content/uploads/OverusedLogos/99gen_circle.jpg",
      name: "Test Company 3",
    },
  ],
}

export { testData }