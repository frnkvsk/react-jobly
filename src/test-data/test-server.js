import { rest } from 'msw';
import { setupServer } from 'msw/node';

let server;

const setServer = async (url, status, message='') => {
  if(server) {
    server.close();
    server.resetHandlers();
  }
  server = await setupServer(
    rest.get(url, (req, res, ctx) => {
      return res(
        ctx.status(status),
        ctx.json(message)
      )
    })
  );
}
  

beforeAll(() => server ? server.listen() : null);
afterAll(() => server ? server.close() : null);
afterEach(() => server ? server.resetHandlers() : null);

export { setServer, server, rest }
