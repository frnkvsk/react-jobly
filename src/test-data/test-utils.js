// test-utils.js
import React from 'react';
import { render } from '@testing-library/react';

import { BrowserRouter } from 'react-router-dom';
import { SearchProvider } from '../context/SearchContext';
import { AuthProvider } from '../context/AuthContext';
import { PageCountProvider } from '../context/PageCountContext';

const AllTheProviders = (page) => {
  return render(
    <BrowserRouter>
      <AuthProvider>
        <SearchProvider>
          <PageCountProvider>
            {page}
          </PageCountProvider>
        </SearchProvider>
      </AuthProvider> 
    </BrowserRouter>
    
  );
}

// re-export everything
export * from '@testing-library/react';

// override render method
export { AllTheProviders as render }
