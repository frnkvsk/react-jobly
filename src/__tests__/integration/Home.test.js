import React from 'react';
import '@testing-library/jest-dom/extend-expect';
import { render, screen } from '../../test-data/test-utils';
import Home from '../../pages/Home';

// smoke and appearance tests
test('renders Home ', async() => {
  render(<Home />);
  
  expect(screen.getByText(/jobly/i)).toBeInTheDocument();

  expect(screen.getByText(/all the jobs in one, convenient place./i)).toBeInTheDocument();

  expect(screen.getByRole('button', {name: /log in \/ sign up/i})).toBeInTheDocument();
});


// snapshot test
test('matches snapshot', () => {
  const { asFragment } = render(<Home />);
  expect(asFragment()).toMatchSnapshot();
});