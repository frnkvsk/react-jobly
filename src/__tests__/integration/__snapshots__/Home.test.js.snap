// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="makeStyles-root-3"
  >
    <main
      class="makeStyles-main-4"
    >
      <h2>
        Jobly
      </h2>
      <p>
        All the jobs in one, convenient place.
      </p>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-label"
        >
          log in / sign up
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </main>
  </div>
</DocumentFragment>
`;
